packages/fast-components-react-msft/app/color-picker.tsx

```diff
@@ -0,0 +1,105 @@
+    backgroundColor: string;
+    accentColor: string;
+}
+
+export interface IColorPickerProps extends IColorConfig {
+    onColorUpdate: (colors: IColorConfig) => void;
+}
+
+export interface IColorPickerManagedClasses {
+    colorPicker: string;
+    colorPicker_label: string;
+}
+
+const styles: ComponentStyles<IColorPickerManagedClasses, IDesignSystem> = {
+    colorPicker: {
+        display: "flex",
+        height: "100%",
+        alignItems: "center"
+    },
+    colorPicker_label: {
+        margin: "0 12px"
+    }
+};
+
+class ColorPicker extends React.Component<IColorPickerProps & IManagedClasses<IColorPickerManagedClasses>, undefined> {
+    constructor(props: IColorPickerProps) {
+        super(props);
+
+        this.foregroundRef = React.createRef();
+        this.backgroundRef = React.createRef();
+        this.accentRef = React.createRef();
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className={this.props.managedClasses.colorPicker}>
+                {this.createColorInput("foreground", this.props.foregroundColor, "foregroundInput", this.foregroundRef)}
+                {this.createColorInput("background", this.props.backgroundColor, "backgroundInput", this.backgroundRef)}
+                {this.createColorInput("accent", this.props.accentColor, "accentInput", this.accentRef)}
+            </div>
+        );
+    }
+
+    private createColorInput(name: string, value: string, id: string, ref: React.RefObject<HTMLInputElement>) {
+        return (
+            <React.Fragment>
+                <label
+                    for={id}
+                    className={this.props.managedClasses.colorPicker_label}
+                >
+                    {name}
+                </label>
+                <input
+                    type="color"
+                    value={this.formatColor(value)}
+                    id={id}
+                    name={name}
+                    onChange={this.handleColorPickerChange}
+                    ref={ref}
+                />
+            </React.Fragment>
+        )
+    }
+
+    private foregroundRef: React.RefObject<HTMLInputElement>;
+    private backgroundRef: React.RefObject<HTMLInputElement>;
+    private accentRef: React.RefObject<HTMLInputElement>;
+
+    /**
+     * Event handler for all color input changes
+     */
+    private handleColorPickerChange = (e: React.FormEvent<HTMLInputElement>) => {
+        const value = e.currentTarget.value;
+        const updatedColorKey: keyof IColorConfig = e.currentTarget === this.foregroundRef.current
+            ? "foregroundColor"
+            : e.currentTarget === this.backgroundRef.current
+            ? "backgroundColor"
+            : "accentColor";
+
+        if (typeof this.props.onColorUpdate === "function") {
+            this.props.onColorUpdate(Object.assign({}, this.props, { [updatedColorKey]: value}));
+        }
+    }
+
+    /**
+     * Ensures that colors are properly formatted
+     */
```

Comment:
If there are input/output parameters I would recommend adding them to the function comments.  Every function, should have comments, inputs, outputs, return statements to help in understanding details for new contributors.

Reply:
In general I would like to defer to TypeScript to describe the inputs and outputs because those are 100% guaranteed to be accurate - comments sometimes lie.

I think comments describing the API can be helpful if the API is hard to type or hard to understand, but I think that explicitly documenting the API in comments as opt-in based on reasoning above. Our type system is pretty accurate and descriptive so I'd like to utilize that as much as possible to avoid redundant and error prone work.